auditlog_security/models/auditlog_rule.py

```diff
@@ -0,0 +1,126 @@
+# Copyright 2021-2024 Therp B.V.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models, tools
+
+
+class AuditlogRule(models.Model):
+    _inherit = "auditlog.rule"
+
+    auditlog_line_access_rule_ids = fields.One2many(
+        "auditlog.line.access.rule", "auditlog_rule_id"
+    )
+    server_action_id = fields.Many2one(
+        "ir.actions.server",
+        "Server Action",
+    )
+    log_selected_fields_only = fields.Boolean(
+        help="Log only the selected fields, to save space avoid large DB data.",
+    )
+
+    @api.model
+    @tools.ormcache("model_name")
+    def _get_field_names_of_rule(self, model_name):
+        """Memory-cached list of fields per rule"""
+        rule = (
+            self.env["auditlog.rule"]
+            .sudo()
+            .search([("model_id.model", "=", model_name)], limit=1)
+        )
+        if rule.auditlog_line_access_rule_ids:
+            return rule.mapped("auditlog_line_access_rule_ids.field_ids.name")
+        return []
+
+    @api.model
+    @tools.ormcache("model_name")
+    def _get_log_selected_fields_only(self, model_name):
+        """Memory-cached translation of model to rule"""
+        rule = (
+            self.env["auditlog.rule"]
+            .sudo()
+            .search([("model_id.model", "=", model_name)], limit=1)
+        )
+        return rule.log_selected_fields_only
+
+    @api.model
+    def get_auditlog_fields(self, model):
+        res = super(AuditlogRule, self).get_auditlog_fields(model)
+        if self._get_log_selected_fields_only(model._name):
+            selected_field_names = self._get_field_names_of_rule(model._name)
+            # we re-use the checks on non-stored fields from super.
+            res = [x for x in selected_field_names if x in res]
+        return res
+
+    def write(self, values):
+        cache_invalidating_fields = [
+            "state",
+            "auditlog_line_access_rule_ids",
+            "log_selected_fields_only",
+        ]
+        if any([field in values.keys() for field in cache_invalidating_fields]):
+            # clear cache for all ormcache methods.
+            self.clear_caches()
+        return super(AuditlogRule, self).write(values)
+
+    @api.onchange("model_id")
+    def onchange_model_id(self):
+        # if model changes we must wipe out all field ids
+        self.auditlog_line_access_rule_ids.unlink()
```

Comment:
```suggestion
        self.auditlog_line_access_rule_ids = False
```

onchange methods should work on the memory, not the database